test/distributed/test_inductor_collectives.py

```diff
@@ -1613,6 +1614,7 @@ def _reorder_communication_preserving_peak_memory(
                 "reorder_for_compute_comm_overlap": True,
                 "reorder_for_compute_comm_overlap_passes": [
                     _reorder_communication_preserving_peak_memory,
```

Comment:
i think it'd be good to add a test that shows that we can move a wait (wait_a) past a collective and past another wait (wait_b), in the case that wait_b and collective are not actually depending on the wait_a